clang-tools-extra/clang-tidy/misc/AlphabeticalDeclOrderCheck.cpp

```diff
@@ -0,0 +1,84 @@
+//===--- AlphabeticalDeclOrderCheck.cpp - clang-tidy ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "AlphabeticalDeclOrderCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace misc {
+
+void AlphabeticalDeclOrderCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(declStmt().bind("stmt"), this);
+}
+
+bool AlphabeticalDeclOrderCheck::checkAlphabeticalOrder(const DeclStmt *MatchedStmt, std::vector<StringRef> &DeclNames, bool &fixDecl) {
+  auto prev_it = MatchedStmt->decl_begin();
+  auto it = MatchedStmt->decl_begin();
+  auto firstVarDecl = dyn_cast<VarDecl>(*MatchedStmt->decl_begin());
+  bool found = true;
+  
+  DeclNames.push_back(firstVarDecl->getName());
+  ++it;
+  
+  while (it != MatchedStmt->decl_end()) {
+    auto varDecl = dyn_cast<VarDecl>(*it);
+    auto prevVarDecl = dyn_cast<VarDecl>(*prev_it);
+
+    if (varDecl->getName() < prevVarDecl->getName())
+      found = false;
+    if (varDecl->getType() != prevVarDecl->getType())
+      fixDecl = false;
+
+    DeclNames.push_back(varDecl->getName());
+    prev_it = it;
+    ++it;
+  }
+  return found;
+}
+
+void AlphabeticalDeclOrderCheck::check(const MatchFinder::MatchResult &Result) {
+  std::vector<StringRef> ret;
+  const auto *MatchedStmt = Result.Nodes.getNodeAs<DeclStmt>("stmt");
+  bool fixDecl = true;
+
+  if (MatchedStmt->isSingleDecl())
+    return;
+
+  auto firstVarDecl = dyn_cast<VarDecl>(*MatchedStmt->decl_begin());
+  if (firstVarDecl ==  NULL) {
```

Comment:
You can just compare like:
```
if (firstVarDecl)
```
If you really want to be explicit, you should check against `nullptr` instead of `NULL`.